sigopt/xgboost/run.py

```diff
@@ -0,0 +1,109 @@
+import platform
+import copy
+
+import xgboost
+# pylint: disable=no-name-in-module
+from xgboost import DMatrix
+
+from ..context import Context
+from ..log_capture import SystemOutputStreamMonitor
+from .. import create_run
+
+DEFAULT_EVALS_NAME = 'Test Set'
+DEFAULT_RUN_OPTIONS = {
+  'log_sys_info': True,
+  'log_stdout': True,
+  'log_stderr': True,
+  'run': None
+}
+XGB_ALIASES = \
+  {
+     'learning_rate': 'eta',
+     'min_split_loss': 'gamma',
+     'reg_lambda': 'lambda',
+     'reg_alpha': 'alpha'
+  }
+
+
+def parse_run_options(run_options):
+  if run_options:
+    assert run_options.keys() <= DEFAULT_RUN_OPTIONS.keys(), 'Unsupported argument inside run_options'
+  run_options_parsed = {**DEFAULT_RUN_OPTIONS, **run_options} if run_options else DEFAULT_RUN_OPTIONS
+  return run_options_parsed
+
+
+class XGBRun:
+
+  def __init__(self, params, dtrain, num_boost_round, evals, run_options):
+    self.params = params
+    self.dtrain = dtrain
+    self.num_boost_round = num_boost_round
+    self.evals = evals
+    self.run_options_parsed = parse_run_options(run_options)
+    self.validation_sets = [(evals, DEFAULT_EVALS_NAME)] if isinstance(evals, DMatrix) else evals
+    self.run = None
+
+  def make_run(self):
+    if self.run_options_parsed['run']:
+      self.run = self.run_options_parsed['run']
```

Comment:
ok. Seems kinda unsafe to just to do this without checking if `run_options.run` is a Run object.

Reply:
kk